### PR TITLE
Support searching multiple files

### PR DESCRIPTION
**Pull request by mathieumg**
See discussion in #11.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,19 +1,25 @@
 'use strict';
 const path = require('path');
+const locatePath = require('locate-path');
 const pathExists = require('path-exists');
 
+const findPath = (dir, files) => locatePath(
+	files.map(file => path.join(dir, file)),
+	{concurrency: 1}
+);
```
Comment (sindresorhus) [inline index.js]: Make it a single line

File: index.js [hunk 2]
```diff
@@ -31,11 +37,17 @@ module.exports.sync = (filename, opts) => {
 	let dir = path.resolve(opts.cwd || '');
 	const root = path.parse(dir).root;
 
+	const filenames = [].concat(filename);
+
+	const findPathSync = files => files
+		.map(name => path.join(dir, name))
+		.find(filePath => pathExists.sync(filePath));
```
Comment (sindresorhus) [inline index.js]: `locatePath.sync()` ;)

File: index.js [hunk 3]
```diff
@@ -31,11 +37,17 @@ module.exports.sync = (filename, opts) => {
...
+
 	// eslint-disable-next-line no-constant-condition
 	while (true) {
-		const fp = path.join(dir, filename);
+		const fp = findPathSync(filenames);
```
Comment (sindresorhus) [inline index.js]: `dir` needs to be path.join'd for each iteration.

Comment (mathieumg) [inline index.js]: It's done in `findPathSync`. I recall I had to take it out of there because the linter complained about my arrow function inside a loop.

**Comment by mathieumg** [comment]
Sorry for the non-grammatical PR title, I just slept 3h in the middle of the night and had just woken up. Thanks for changing it! I made the requested changes. I imagine updating the README is next on the list?

File: index.js (hunk 3 above)
Comment (mathieumg) [inline index.js]: Alright, I changed it.

File: index.js [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
 
+	const filenames = [].concat(filename);
+
 	return new Promise(resolve => {
 		(function find(dir) {
-			const fp = path.join(dir, filename);
-
-			pathExists(fp).then(exists => {
-				if (exists) {
+			findPath(dir, filenames).then(fp => {
```
Comment (sindresorhus) [inline index.js]: I just added a `cwd` option to `locatePath`, so you can now use it directly inline without a wrapper. (Sorry about all the back-and-forth).

Make sure to bump the version to `^1.1.0`.

**Comment by sindresorhus** [comment]
> I imagine updating the README is next on the list?

Yes

File: index.js (hunk 4 above)
Comment (mathieumg) [inline index.js]: Before I add the post-find path joining logic myself here, is it by design that `locate-path` always only return the filename?

**Comment by sholladay** [comment]
The fact that [locate-path](https://github.com/sindresorhus/locate-path) needs `{concurrency: 1}` to preserve the search order is unfortunate.

I don't think we want to wait for the result of the first path to begin checking the second (and so forth). We had been talking about this I/O optimization in #11.

Other than that, I think this is great.

**Comment by mathieumg** [comment]
AppVeyor seems to have choked on the 4.x build. I thought they added a way to relaunch builds a while ago, or was it just for repo owners? In any event, I imagine I could force-push to retrigger the build.

**Comment by sindresorhus** [comment]
> The fact that locate-path needs {concurrency: 1} to preserve the search order is unfortunate.

Agreed. When I built it I thought it was the only way. Thinking about it now, it should be possible to start all of the path checks concurrently, then when one matches, ensure that the elements before has finished and didn't match, otherwise wait for those. Help/ideas appriciated on how to solve that in [`p-locate`](https://github.com/sindresorhus/p-locate). It's late here now and I can't wrap my head around it. Should probably be behind an option called `preserveOrder`.

**Comment by mathieumg** [comment]
> it should be possible to start all of the path checks concurrently, then when one matches, ensure that the elements before has finished and didn't match, otherwise wait for those.

It's pretty much what my previous implementation here did. Thanks @sholladay for bringing that back up. Whatever avenue is taken, that `preserveOrder` option should probably be on by default since `p-locate` purports to be "an async version of `Array#find`."

**Comment by sindresorhus** [comment]
@mathieumg Ah, sorry I missed that. I guess I didn't think of that use-case when I first made `p-locate`. It seems like your implementation first starts all the path check promises and then serially iterates through them checking whether they're done and matching. I'll work on it after some sleep.

**Comment by sindresorhus** [comment]
It was a bit hard to satisfy the constraints of supporting both concurrency limit and a `preserveOrder` option, but I managed to get something thrown together finally: https://github.com/sindresorhus/p-locate/pull/1

**Comment by sindresorhus** [comment]
locate-path@2 that runs concurrently, but preserves order, is published.

**Comment by sindresorhus** [comment]
@sholladay Can you take a final look and 👍  it?

**Comment by mathieumg** [comment]
I just realized my commit message doesn't mention the right module name. Will the commits be squashed anyway?

**Comment by sindresorhus** [comment]
> Will the commits be squashed anyway?

Yes

**Comment by sholladay** [comment]
This LGTM. 👍 

I'll repeat one thought from #11 for posterity.

> With the array example, I'm not sure it's clear that the semantics are _some_ and not _every_. And if this were to be implemented, I feel like people will end up asking for both.

In other words, I might want to search for `package.json` but only stop when it's also next to a `.git` directory. Various new use cases might seem in scope if we begin handling multiple files. Just something to keep in mind and define the limits of for this module vs making a new project.

**Comment by mathieumg** [comment]
> Various new use cases might seem in scope if we begin handling multiple files.

Indeed.

> I feel like people will end up asking for both.

I imagine those could be tackled as they come in, it would otherwise be harder to do a "one size fits all" with spread out and blurry "requirements".

**Comment by sindresorhus** [comment]
Thank you @mathieumg :)

**Comment by sindresorhus** [comment]
> In other words, I might want to search for package.json but only stop when it's also next to a .git directory.

I think the current use-case is a lot more common than this though.

> Various new use cases might seem in scope if we begin handling multiple files. Just something to keep in mind and define the limits of for this module vs making a new project.

Sure, but we can tackle that when/if requests come in.

**Pull request merged**